src/main/java/com/compfest/sea/delivery/user/UserDelivery.java

```diff
@@ -37,26 +38,43 @@ public UserDelivery(@Qualifier("UserUsecaseImpl") UserUsecase usecase) {
     @PostMapping
     public User register(@RequestBody InsertUserRequestPayload payload) {
         User convertedFromPayload = UserAdapter.convertInsertUserRequestPayloadToUser(payload);
+        User user = userUsecase.findUserByEmail(convertedFromPayload.getEmail());
```

Comment:
hmm sorryy,

kemarin jadinya diganti yang ada Role nya kah...(?)
mungkin ini perlu validasi exist emailnya sebagai Role apa...

kalau tidak salah di Adli punya yang di`unique`-in ada email dan role nya...
`@UniqueConstraint(columnNames = {"email", "role"})`

Reply:
iya juga nanti ku coba buat